qsimcirq/qsim_circuit.py

```diff
@@ -104,6 +106,27 @@ def _cirq_gate_kind(gate):
   return None
 
 
+def _control_details(gate: cirq.ops.ControlledGate, qubits):
+  control_qubits = []
+  control_values = []
+  # TODO: support qudit control
+  for i, cvs in enumerate(gate.control_values):
+    if 0 in cvs and 1 in cvs:
+      # This qubit does not affect control.
+      continue
+    elif 0 not in cvs and 1 not in cvs:
+      # This gate will never trigger.
+      raise ValueError(f'Gate has no valid control value: {gate}')
+    else:
```

Comment:
Don't really need this `else` since you will either continue or throw an exception beforehand. Can this conditional logic be simplified in general ?

Reply:
Not sure how much simplification is possible here, since all four cases need to be checked. Covering the `0 in cvs` cases first splits up the only-0 and only-1 cases, which share behavior. I can remove the else, though.